src/transmission_manager.rs

```diff
@@ -44,36 +50,38 @@ fn evaluate_transmission(context: &mut Context, contact_id: PersonId, _transmitt
 
 fn infection_attempt(
     context: &mut Context,
-    transmitee_id: PersonId,
-    gi: f64,
-    num_infection_attempts_remaining: f64,
+    transmitter_id: PersonId,
+    num_infection_attempts_remaining: usize,
     next_infection_time_unif: f64,
 ) -> Result<(), IxaError> {
-    // if the person still has people left to infect
-    if num_infection_attempts_remaining >= 1.0 {
-        // this is a method from a trait extention implemented in the contact manager
-        // as long as this method returns a contact id, it can use any underlying sampling strategy
-        let contact_id = context.get_contact(transmitee_id)?;
-        // evaluate transmission is its own function because there will be
-        // intervention-based logic there eventually
-        // if there are no contacts, to infect, do nothing
+    // Only attempt the infection if the person has infection attempts remaining.
+    if num_infection_attempts_remaining >= 1 {
+        // This is a method from a trait extension implemented in `mod contact`.
+        // As long as the method returns a contact id, it can use any underlying sampling strategy
+        // to obtain that contact, and that strategy can be separately implemented in `mod contact`
+        // without changing the logic here.
+        let contact_id = context.get_contact(transmitter_id)?;
+
+        // We evvaluate transmission in its own function because there will be eventually
+        // be intervention-based logic there.
+        // If there are no contacts to infect, do nothing.
         if let Some(contact_id) = contact_id {
-            evaluate_transmission(context, contact_id, transmitee_id);
+            evaluate_transmission(context, contact_id, transmitter_id);
         }
-        // schedule the subsequent infection attempt for this infected agent,
-        // which happens at a greater value of the GI
+
+        // Schedule the next infection attempt for this infected agent.
+        // The next infection attempt happens at a greater value of the GI than the last infection attempt.
         schedule_next_infection_attempt(
             context,
-            transmitee_id,
-            gi,
-            num_infection_attempts_remaining - 1.0,
+            transmitter_id,
+            num_infection_attempts_remaining - 1,
             next_infection_time_unif,
         );
     }
-    // if no more infection attempts remaining, set the person to recovered
+    // If there are no more infection attempts remaining, set the person to recovered.
```

Comment:
I would put this case first, and then you can return right away, rather than having an else.